starknet/compiler_test.go

```diff
@@ -37,4 +38,265 @@ func TestCompile(t *testing.T) {
 		require.NoError(t, err)
 		assert.Equal(t, expectedCompiled.Hash(), gotCompiled.Hash())
 	})
+
+	t.Run("declare cairo2 class", func(t *testing.T) {
+		// tests https://github.com/NethermindEth/juno/issues/1748
+		_, err := starknet.Compile(&starknet.SierraDefinition{
+			EntryPoints: starknet.SierraEntryPoints{
```

Comment:
It might be a good idea to have this "test data" deserialize from external json file? (if it's easy enough)
(or at least "program" part 👇 )
Containing it here make this test bloated. WDYT?

Reply:
done